In [1]:
from utils import max_features, maxlen, embed_size_fastText, embed_size_glove, embed_size_glove_twitter
from utils import earlystop, batch_size, epochs, run_5fold_2channels
from model import get_multiChannel_CNN_model
import pickle

/home/ec2-user/anaconda3/envs/tensorflow_p36/lib/python3.6/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
Using TensorFlow backend.
/home/ec2-user/anaconda3/envs/tensorflow_p36/lib/python3.6/site-packages/matplotlib/__init__.py:962: UserWarning: Duplicate key in file "/home/ec2-user/.config/matplotlib/matplotlibrc", line #2
  (fname, cnt))
/home/ec2-user/anaconda3/envs/tensorflow_p36/lib/python3.6/site-packages/matplotlib/__init__.py:962: UserWarning: Duplicate key in file "/home/ec2-user/.config/matplotlib/matplotlibrc", line #3
  (fname, cnt))
/home/ec2-user/anaconda3/envs/tensorflow_p36/lib/python3.6/site-packages/sklearn/cross_validation.py:41: DeprecationWarning: This module was deprecated in version 0.18 in favor of the model_selection module into whic

In [2]:
fastText = pickle.load(open("input/fastText.300d.pickle", "rb"))
glove = pickle.load(open("input/glove.300d.pickle", "rb"))

In [3]:
x_train = pickle.load(open("input/x_train_small.pickle", "rb"))
y_train = pickle.load(open("input/y_train_small.pickle", "rb"))

### baseline performance

- fastText single layer

acc: 97.7272242339% (+/- 0.0262575199%)

roc_auc: 99.3181312003% (+/- 0.0253788972%)

- glove single layer

acc: 97.7449803026% (+/- 0.0835450620%)

roc_auc: 99.3592604342% (+/- 0.0654025651%)

In [4]:
params = {
    "maxlen": maxlen,
    "max_features": max_features,
    "embed_size1": embed_size_fastText,
    "embedding_matrix1": fastText,
    "dropout1": 0.1,
    "num_filter1": 500,
    "kernel_size1": 9,
    "reg1": 0,
    "embed_size2": embed_size_glove,
    "embedding_matrix2": glove,
    "dropout2": 0.2,
    "num_filter2": 500,
    "kernel_size2": 7,
    "reg2": 0
}  

## check dense_units

In [5]:
units = [128, 256, 512, 768]

In [6]:
for u in units:
    print ("\n********************u:{}********************".format(u))
    params["dense_units"] = u
    run_5fold_2channels([x_train, x_train], y_train, get_multiChannel_CNN_model, **params)


********************u:128********************
Instructions for updating:
Please switch to tf.metrics.auc. Note that the order of the labels and predictions arguments has been switched.
Epoch 1/50
 - 24s - loss: 0.1516 - acc: 0.9614 - auc_roc: 0.7348
Epoch 2/50
 - 22s - loss: 0.0667 - acc: 0.9802 - auc_roc: 0.8638
Epoch 3/50
 - 22s - loss: 0.0333 - acc: 0.9912 - auc_roc: 0.9189
Epoch 4/50
 - 22s - loss: 0.0189 - acc: 0.9966 - auc_roc: 0.9464
Epoch 5/50
 - 22s - loss: 0.0138 - acc: 0.9984 - auc_roc: 0.9607
Epoch 6/50
 - 22s - loss: 0.0109 - acc: 0.9991 - auc_roc: 0.9689
Epoch 7/50
 - 22s - loss: 0.0103 - acc: 0.9992 - auc_roc: 0.9740
Epoch 8/50
 - 22s - loss: 0.0097 - acc: 0.9993 - auc_roc: 0.9775
Epoch 9/50
 - 22s - loss: 0.0084 - acc: 0.9994 - auc_roc: 0.9800
Epoch 10/50
 - 22s - loss: 0.0068 - acc: 0.9992 - auc_roc: 0.9820
Epoch 11/50
 - 22s - loss: 0.0039 - acc: 0.9996 - auc_roc: 0.9837
Epoch 12/50
 - 22s - loss: 0.0025 - acc: 0.9997 - auc_roc: 0.9853
Epoch 13/50
 - 22s - loss: 0.00

Epoch 17/50
 - 22s - loss: 0.0096 - acc: 0.9994 - auc_roc: 0.9868
Epoch 18/50
 - 22s - loss: 0.0099 - acc: 0.9994 - auc_roc: 0.9872
Epoch 19/50
 - 22s - loss: 0.0100 - acc: 0.9994 - auc_roc: 0.9875
Epoch 20/50
 - 22s - loss: 0.0097 - acc: 0.9994 - auc_roc: 0.9878
Epoch 21/50
 - 22s - loss: 0.0096 - acc: 0.9994 - auc_roc: 0.9880
Epoch 22/50
 - 22s - loss: 0.0098 - acc: 0.9994 - auc_roc: 0.9883
Epoch 23/50
 - 22s - loss: 0.0097 - acc: 0.9994 - auc_roc: 0.9885
Epoch 24/50
 - 22s - loss: 0.0096 - acc: 0.9994 - auc_roc: 0.9887
Epoch 25/50
 - 22s - loss: 0.0096 - acc: 0.9994 - auc_roc: 0.9888
Epoch 26/50
 - 22s - loss: 0.0094 - acc: 0.9994 - auc_roc: 0.9890
Epoch 27/50
 - 22s - loss: 0.0096 - acc: 0.9994 - auc_roc: 0.9891
Epoch 28/50
 - 22s - loss: 0.0099 - acc: 0.9993 - auc_roc: 0.9893
Epoch 29/50
 - 22s - loss: 0.4456 - acc: 0.9693 - auc_roc: 0.9846
Epoch 30/50
 - 22s - loss: 0.5974 - acc: 0.9629 - auc_roc: 0.9683
Epoch 31/50
 - 22s - loss: 0.5798 - acc: 0.9638 - auc_roc: 0.9528
Epoch 32/5

Epoch 39/50
 - 22s - loss: 0.0010 - acc: 0.9998 - auc_roc: 0.9966
Epoch 40/50
 - 22s - loss: 9.5435e-04 - acc: 0.9999 - auc_roc: 0.9967
Epoch 41/50
 - 22s - loss: 9.7535e-04 - acc: 0.9999 - auc_roc: 0.9968
Epoch 42/50
 - 22s - loss: 0.0021 - acc: 0.9996 - auc_roc: 0.9968
Epoch 43/50
 - 22s - loss: 0.0054 - acc: 0.9991 - auc_roc: 0.9968
Epoch 44/50
 - 22s - loss: 0.0053 - acc: 0.9988 - auc_roc: 0.9969
Epoch 45/50
 - 22s - loss: 0.0098 - acc: 0.9976 - auc_roc: 0.9968
Epoch 46/50
 - 22s - loss: 0.0037 - acc: 0.9990 - auc_roc: 0.9968
Epoch 47/50
 - 22s - loss: 0.0046 - acc: 0.9989 - auc_roc: 0.9969
Epoch 48/50
 - 22s - loss: 0.0013 - acc: 0.9996 - auc_roc: 0.9969
Epoch 49/50
 - 22s - loss: 0.0017 - acc: 0.9995 - auc_roc: 0.9970
Epoch 50/50
 - 22s - loss: 0.0018 - acc: 0.9996 - auc_roc: 0.9970
acc: 97.6078540993%
auc_roc: 99.6612222319%
acc: 97.7460335099% (+/- 0.1470031126%)
roc_auc: 97.5354285869% (+/- 4.1264994645%)

********************u:256********************
Epoch 1/50
 - 23s - loss:

 - 22s - loss: 0.0021 - acc: 0.9998 - auc_roc: 0.9796
Epoch 9/50
 - 22s - loss: 0.0014 - acc: 0.9999 - auc_roc: 0.9828
Epoch 10/50
 - 22s - loss: 0.0013 - acc: 0.9999 - auc_roc: 0.9851
Epoch 11/50
 - 22s - loss: 0.0011 - acc: 0.9999 - auc_roc: 0.9869
Epoch 12/50
 - 22s - loss: 0.0015 - acc: 0.9998 - auc_roc: 0.9884
Epoch 13/50
 - 22s - loss: 9.8911e-04 - acc: 0.9999 - auc_roc: 0.9895
Epoch 14/50
 - 22s - loss: 8.4160e-04 - acc: 0.9999 - auc_roc: 0.9905
Epoch 15/50
 - 22s - loss: 4.8274e-04 - acc: 1.0000 - auc_roc: 0.9913
Epoch 16/50
 - 22s - loss: 6.0174e-04 - acc: 1.0000 - auc_roc: 0.9920
Epoch 17/50
 - 22s - loss: 6.2361e-04 - acc: 0.9999 - auc_roc: 0.9926
Epoch 18/50
 - 22s - loss: 7.0785e-04 - acc: 0.9999 - auc_roc: 0.9931
Epoch 19/50
 - 22s - loss: 5.7480e-04 - acc: 1.0000 - auc_roc: 0.9935
Epoch 20/50
 - 22s - loss: 4.1862e-04 - acc: 1.0000 - auc_roc: 0.9939
Epoch 21/50
 - 22s - loss: 5.1183e-04 - acc: 1.0000 - auc_roc: 0.9943
Epoch 22/50
 - 22s - loss: 7.4988e-04 - acc: 0.9999 -

Epoch 30/50
 - 22s - loss: 0.5827 - acc: 0.9638 - auc_roc: 0.8462
Epoch 31/50
 - 22s - loss: 0.5827 - acc: 0.9638 - auc_roc: 0.8345
Epoch 32/50
 - 22s - loss: 0.5827 - acc: 0.9638 - auc_roc: 0.8235
Epoch 33/50
 - 22s - loss: 0.5827 - acc: 0.9638 - auc_roc: 0.8130
Epoch 34/50
 - 22s - loss: 0.5827 - acc: 0.9638 - auc_roc: 0.8034
Epoch 35/50
 - 22s - loss: 0.5827 - acc: 0.9638 - auc_roc: 0.7944
Epoch 36/50
 - 22s - loss: 0.5827 - acc: 0.9638 - auc_roc: 0.7857
Epoch 37/50
 - 22s - loss: 0.5827 - acc: 0.9638 - auc_roc: 0.7777
Epoch 38/50
 - 22s - loss: 0.5827 - acc: 0.9638 - auc_roc: 0.7700
Epoch 39/50
 - 22s - loss: 0.5827 - acc: 0.9638 - auc_roc: 0.7627
Epoch 40/50
 - 22s - loss: 0.5827 - acc: 0.9638 - auc_roc: 0.7558
Epoch 41/50
 - 22s - loss: 0.5827 - acc: 0.9638 - auc_roc: 0.7492
Epoch 42/50
 - 22s - loss: 0.5827 - acc: 0.9638 - auc_roc: 0.7431
Epoch 43/50
 - 22s - loss: 0.5827 - acc: 0.9638 - auc_roc: 0.7372
Epoch 44/50
 - 22s - loss: 0.5827 - acc: 0.9638 - auc_roc: 0.7317
Epoch 45/5

Epoch 49/50
 - 22s - loss: 7.6348e-04 - acc: 0.9999 - auc_roc: 0.9957
Epoch 50/50
 - 22s - loss: 7.5622e-04 - acc: 0.9999 - auc_roc: 0.9958
acc: 97.7182524694%
auc_roc: 99.5341509506%
Epoch 1/50
 - 23s - loss: 0.1604 - acc: 0.9609 - auc_roc: 0.6871
Epoch 2/50
 - 22s - loss: 0.0672 - acc: 0.9776 - auc_roc: 0.8404
Epoch 3/50
 - 22s - loss: 0.0350 - acc: 0.9881 - auc_roc: 0.9067
Epoch 4/50
 - 22s - loss: 0.0168 - acc: 0.9945 - auc_roc: 0.9396
Epoch 5/50
 - 22s - loss: 0.0076 - acc: 0.9979 - auc_roc: 0.9577
Epoch 6/50
 - 22s - loss: 0.0045 - acc: 0.9992 - auc_roc: 0.9684
Epoch 7/50
 - 22s - loss: 0.0024 - acc: 0.9997 - auc_roc: 0.9751
Epoch 8/50
 - 22s - loss: 0.0019 - acc: 0.9998 - auc_roc: 0.9798
Epoch 9/50
 - 22s - loss: 0.0017 - acc: 0.9999 - auc_roc: 0.9830
Epoch 10/50
 - 22s - loss: 0.0011 - acc: 0.9999 - auc_roc: 0.9854
Epoch 11/50
 - 22s - loss: 0.0012 - acc: 0.9999 - auc_roc: 0.9872
Epoch 12/50
 - 22s - loss: 0.0012 - acc: 0.9999 - auc_roc: 0.9887
Epoch 13/50
 - 22s - loss: 9.1135

 - 22s - loss: 0.0019 - acc: 0.9996 - auc_roc: 0.9910
Epoch 20/50
 - 22s - loss: 0.0302 - acc: 0.9944 - auc_roc: 0.9911
Epoch 21/50
 - 22s - loss: 0.0238 - acc: 0.9933 - auc_roc: 0.9907
Epoch 22/50
 - 22s - loss: 0.0067 - acc: 0.9982 - auc_roc: 0.9909
Epoch 23/50
 - 22s - loss: 0.0060 - acc: 0.9985 - auc_roc: 0.9912
Epoch 24/50
 - 22s - loss: 0.0020 - acc: 0.9997 - auc_roc: 0.9916
Epoch 25/50
 - 22s - loss: 0.0016 - acc: 0.9997 - auc_roc: 0.9919
Epoch 26/50
 - 22s - loss: 0.0013 - acc: 0.9998 - auc_roc: 0.9922
Epoch 27/50
 - 22s - loss: 0.0018 - acc: 0.9997 - auc_roc: 0.9926
Epoch 28/50
 - 22s - loss: 0.0011 - acc: 0.9999 - auc_roc: 0.9929
Epoch 29/50
 - 22s - loss: 9.6146e-04 - acc: 0.9999 - auc_roc: 0.9931
Epoch 30/50
 - 22s - loss: 8.8257e-04 - acc: 0.9999 - auc_roc: 0.9934
Epoch 31/50
 - 22s - loss: 0.0010 - acc: 0.9999 - auc_roc: 0.9936
Epoch 32/50
 - 22s - loss: 0.0013 - acc: 0.9998 - auc_roc: 0.9938
Epoch 33/50
 - 22s - loss: 8.7724e-04 - acc: 0.9999 - auc_roc: 0.9940
Epoch 34/5

Epoch 39/50
 - 22s - loss: 5.7002e-04 - acc: 0.9999 - auc_roc: 0.9959
Epoch 40/50
 - 22s - loss: 7.6000e-04 - acc: 0.9999 - auc_roc: 0.9960
Epoch 41/50
 - 22s - loss: 6.9369e-04 - acc: 0.9999 - auc_roc: 0.9961
Epoch 42/50
 - 22s - loss: 5.8293e-04 - acc: 0.9999 - auc_roc: 0.9962
Epoch 43/50
 - 22s - loss: 7.9298e-04 - acc: 0.9999 - auc_roc: 0.9963
Epoch 44/50
 - 22s - loss: 0.0022 - acc: 0.9996 - auc_roc: 0.9963
Epoch 45/50
 - 22s - loss: 0.0546 - acc: 0.9926 - auc_roc: 0.9961
Epoch 46/50
 - 22s - loss: 0.0328 - acc: 0.9941 - auc_roc: 0.9953
Epoch 47/50
 - 22s - loss: 0.0110 - acc: 0.9975 - auc_roc: 0.9952
Epoch 48/50
 - 22s - loss: 0.0047 - acc: 0.9991 - auc_roc: 0.9952
Epoch 49/50
 - 22s - loss: 0.0038 - acc: 0.9995 - auc_roc: 0.9953
Epoch 50/50
 - 22s - loss: 0.0020 - acc: 0.9996 - auc_roc: 0.9953
acc: 97.7391382357%
auc_roc: 99.4918249454%
Epoch 1/50
 - 23s - loss: 0.1678 - acc: 0.9572 - auc_roc: 0.7053
Epoch 2/50
 - 22s - loss: 0.0685 - acc: 0.9775 - auc_roc: 0.8377
Epoch 3/50
 - 

Epoch 10/50
 - 22s - loss: 0.0015 - acc: 0.9999 - auc_roc: 0.9818
Epoch 11/50
 - 22s - loss: 0.0012 - acc: 0.9999 - auc_roc: 0.9841
Epoch 12/50
 - 22s - loss: 0.0012 - acc: 0.9999 - auc_roc: 0.9859
Epoch 13/50
 - 22s - loss: 0.0012 - acc: 0.9999 - auc_roc: 0.9873
Epoch 14/50
 - 22s - loss: 0.0010 - acc: 0.9999 - auc_roc: 0.9885
Epoch 15/50
 - 22s - loss: 0.0011 - acc: 0.9999 - auc_roc: 0.9895
Epoch 16/50
 - 22s - loss: 0.0014 - acc: 0.9998 - auc_roc: 0.9903
Epoch 17/50
 - 22s - loss: 0.0027 - acc: 0.9994 - auc_roc: 0.9910
Epoch 18/50
 - 22s - loss: 0.0027 - acc: 0.9995 - auc_roc: 0.9916
Epoch 19/50
 - 22s - loss: 0.0149 - acc: 0.9965 - auc_roc: 0.9920
Epoch 20/50
 - 22s - loss: 0.0210 - acc: 0.9944 - auc_roc: 0.9917
Epoch 21/50
 - 22s - loss: 0.0059 - acc: 0.9983 - auc_roc: 0.9919
Epoch 22/50
 - 22s - loss: 0.0042 - acc: 0.9990 - auc_roc: 0.9923
Epoch 23/50
 - 22s - loss: 0.0021 - acc: 0.9996 - auc_roc: 0.9926
Epoch 24/50
 - 22s - loss: 0.0021 - acc: 0.9996 - auc_roc: 0.9930
Epoch 25/5

##### result

- 128: 

acc: 97.7460335099% (+/- 0.1470031126%)

roc_auc: 97.5354285869% (+/- 4.1264994645%)

- 256:

acc: 97.4942727921% (+/- 0.7053694075%)

roc_auc: 92.2276879419% (+/- 11.2570677909%)

- 512:

acc: 97.7585674088% (+/- 0.1304108819%)

roc_auc: 99.5166337291% (+/- 0.1620516927%)

- 768:

acc: 97.7648326617% (+/- 0.1258659665%)

roc_auc: 99.5579392558% (+/- 0.0663592110%)

In [4]:
params = {
    "maxlen": maxlen,
    "max_features": max_features,
    "embed_size1": embed_size_fastText,
    "embedding_matrix1": fastText,
    "dropout1": 0.1,
    "num_filter1": 500,
    "kernel_size1": 9,
    "reg1": 0,
    "embed_size2": embed_size_glove,
    "embedding_matrix2": glove,
    "dropout2": 0.2,
    "num_filter2": 500,
    "kernel_size2": 7,
    "reg2": 0,
    "dense_units": 768
}  